torchtitan/models/norms.py

```diff
@@ -87,17 +89,26 @@ class RMSNorm(nn.Module):
 
     """
 
-    def __init__(self, dim: int, eps: float = 1e-6):
+    def __init__(self, dim: int, eps: float = 1e-6, compile: bool = False):
         super().__init__()
         self.eps = eps
         self.weight = nn.Parameter(torch.ones(dim))
+        self.rmsnorm_fn = (
+            torch.compile(self.compute_rmsnorm, fullgraph=True)
```

Comment:
Does nested `torch.compile` work? E.g., `torch.compile` each transformer block with a `torch.compile`'d `RMSNorm` inside?

Reply:
From my experiment, it works.

Reply:
very cool
is it equivalent perf as `torch.compile` each transformer block but without `torch.compile` `RMSNorm` inside?

Reply:
Here's the result on 8b with `torch.compile` each transformer block. I do see very tiny improvement from the nested compilation. @anijain2305 any insights on this?

rmsnorm:
<img width="754" alt="image" src="https://github.com/pytorch/torchtitan/assets/150487191/b16d6550-33f5-40d4-b927-863115f145c0">

compiled_rmsnorm:
<img width="752" alt="image" src="https://github.com/pytorch/torchtitan/assets/150487191/396fc46b-1e1e-484e-85f3-f61aa50f8d65">

Reply:
memory decreased too 😮

Reply:
I think it likely comes from the outside norm not in any TransformerBlock. @awgu

Reply:
Have you tested with AC? I recall that the nested compilation may have issues with AC, but I could be wrong.

Reply:
It used to fail with full AC and selective op AC, but now the issues disappeared. @fegin